spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -146,13 +149,14 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 
 	@Override
 	public List<String> getServices() {
-		return adapter.apply(client).stream().map(s -> s.getMetadata().getName()).toList();
+		return services(properties, client, namespaceProvider, filter(), null, "fabric8-discovery").stream()
```

Comment:
don't use "adapter", delegate to `services`. This will also be used by the new functionality of external-name services